### DISCUSSION TITLE
Upload plugin. Image uploaded but got error message

### DISCUSSION THREAD

**Discussion by VrbovskyM** [Alex-D/Trumbowyg#1369] [Q&A] [answered]
I can't seem to figure out how to insert the image for whole day :D
I can upload an image to the server with the the plugin, but the editor still show error and doesnt insert it into editor.

Here is my .php file where I handle the post request

```php
$target_dir = "uploaded/";
$target_file = $target_dir . basename($_FILES[$fileFieldName]["name"]);
if (move_uploaded_file($_FILES[$fileFieldName]["tmp_name"], $target_file)) {
    $data = array(
        'success' => true,
        'file' => $target_file,
    );
  } else {
    $data = array(
        'success' => false,
    );
  }
echo json_encode($data);
```

I tried using 'file' and 'url' in the $data array.  Same outcome.

.js file where I edit the trumbowyg

```js
$('#edit')
.trumbowyg({
    btnsDef: {
        // Create a new dropdown
        image: {
            dropdown: ['insertImage', 'upload'],
            ico: 'insertImage'
        }
    },
    // Redefine the button pane
    btns: [
        ['viewHTML'],
        ['fontfamily'],
        ['fontsize'],
        ['formatting'],
        ['strong', 'em', 'del'],
        ['superscript', 'subscript'],
        ['indent', 'outdent'],
        ['lineheight'],
        ['link'],
        ['image'], 
        ['justifyLeft', 'justifyCenter', 'justifyRight', 'justifyFull'],
        ['unorderedList', 'orderedList'],
        ['horizontalRule'],
        ['removeformat'],
        ['fullscreen'],
        ['foreColor', 'backColor'],
    ],
    plugins: {
        // Add imagur parameters to upload plugin for demo purposes
        upload: {
            serverPath: 'upload.php',
            fileFieldName: 'image',
        }
    }
});
```

upload.php handles post request

![image](https://user-images.githubusercontent.com/100531156/216833316-d979a386-f4f4-409e-8b7e-531301e72e3d.png)

I added some console.logs to the trumbowyg.upload.js and its always going to the `error: trumbowyg.o.plugins.upload.error || function () {` callback. Even tho Im returning json with success: true

**Comment by Alex-D** [comment, marked as answer]
Hey!

I think your Content-Type is not set to `application/json` in the response :/

**Reply by VrbovskyM** [reply]
Hi. I resolved my issue.
In my upload.php file I had echo commands in the if statements. So basically the problem was that I had echoing more things than just the json
